selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/filewatcher/CheckFileWatcherExcludeFeatureTest.java

```diff
@@ -53,140 +58,146 @@ public void setUp() throws Exception {
-        ProjectExplorer.FileWatcherExcludeOperations.ADD_TO_FILE_WATCHER_EXCLUDES);
-    editor.selectTabByName(FILE_WATCHER_IGNORE_FILE_NAME);
-    editor.waitTextIntoEditor(FILE_NAME_FOR_EXCLUDE);
+    // Add the file to exclude and check that the file content in the Editor is not changed
+    doFileWatcherExcludeOperation(pathToExcludedFile, ADD_TO_FILE_WATCHER_EXCLUDES);
 
     terminal.selectTerminalTab();
     terminal.waitTerminalConsole();
     terminal.waitTerminalIsNotEmpty();
     terminal.typeIntoTerminal("cd /projects/" + PROJECT_NAME + Keys.ENTER);
     terminal.waitExpectedTextIntoTerminal("/projects/" + PROJECT_NAME);
-    terminal.typeIntoTerminal("pwd > " + FILE_NAME_FOR_EXCLUDE + Keys.ENTER);
-
-    editor.selectTabByName(FILE_NAME_FOR_EXCLUDE);
+    terminal.typeIntoTerminal("pwd > " + fileNameForExluding + Keys.ENTER);
+    editor.selectTabByName(fileNameForExluding);
     editor.waitTextNotPresentIntoEditor("/projects/" + PROJECT_NAME);
-    editor.clickOnCloseFileIcon(FILE_NAME_FOR_EXCLUDE);
-    projectExplorer.openItemByVisibleNameInExplorer(FILE_NAME_FOR_EXCLUDE);
+
+    // Close file and open again and check that the file content is changed
+    editor.clickOnCloseFileIcon(fileNameForExluding);
+    projectExplorer.openItemByVisibleNameInExplorer(fileNameForExluding);
     editor.waitActiveEditor();
     editor.waitTextIntoEditor("/projects/" + PROJECT_NAME);
+    doFileWatcherExcludeOperation(pathToExcludedFile, REMOVE_FROM_FILE_WATCHER_EXCLUDES);
 
-    doFileWatcherExcludeOperation(
-        PATH_FOR_EXCLUDED_FILE,
-        ProjectExplorer.FileWatcherExcludeOperations.REMOVE_FROM_FILE_WATCHER_EXCLUDES);
-    editor.selectTabByName(FILE_WATCHER_IGNORE_FILE_NAME);
-    editor.waitTextNotPresentIntoEditor(FILE_NAME_FOR_EXCLUDE);
+    editor.closeAllTabsByContextMenu();
   }
 
-  @Test(priority = 3)
-  public void checkFeatureAfterExcludingFolder() {
+  @Test
+  public void testFeatureAfterExcludingFolder() {
+    String fileNameForExcluding = "AppController.java";
+    String pathToJavaFile = "/projects/" + PROJECT_NAME + "/src/main/java/org/eclipse/qa/examples";
+
+    // Exclude 'src' folder
     doFileWatcherExcludeOperation(
-        PROJECT_NAME + "/" + FOLDER_NAME_FOR_EXCLUDE,
-        ProjectExplorer.FileWatcherExcludeOperations.ADD_TO_FILE_WATCHER_EXCLUDES);
-    projectExplorer.openItemByVisibleNameInExplorer("AppController.java");
+        PROJECT_NAME + "/" + FOLDER_NAME_FOR_EXCLUDE, ADD_TO_FILE_WATCHER_EXCLUDES);
+
+    projectExplorer.waitItem(PROJECT_NAME);
+    projectExplorer.openItemByVisibleNameInExplorer(fileNameForExcluding);
     editor.waitActiveEditor();
 
+    // Change content of file from the excluded folder
     terminal.selectTerminalTab();
     terminal.waitTerminalConsole();
     terminal.waitTerminalIsNotEmpty();
+    terminal.typeIntoTerminal("cd " + pathToJavaFile + Keys.ENTER);
+    terminal.waitExpectedTextIntoTerminal(pathToJavaFile);
     terminal.typeIntoTerminal(
```

Comment:
It's hard to read the multi line string. it's better to rewrote it by using **String.format()** method.